src/app/static/src/app/components/genericChart/GenericChart.vue

```diff
@@ -252,7 +295,46 @@
                 return !this.chartData ||
                     !Object.values(this.chartData).some(e => e.length);
 
-            }
+            },
+            allDistinctColumnValues() {
+                const subplots = this.chartMetadata.subplots;
+                if (subplots && this.chartData) {
+                    return this.chartData["data"].reduce((distinct: string[], row: any) => {
+                        const distinctVal = row[subplots.distinctColumn];
+                        return distinct.includes(distinctVal) ? distinct : [...distinct, distinctVal]
+                    }, []);
+                } else {
+                    return null;
+                }
+            },
+            pageDistinctColumnValues() {
+                if (this.chartData && this.chartMetadata.subplots && this.allDistinctColumnValues) {
+                    const subplotsPerPage = this.chartMetadata.subplots.subplotsPerPage;
+                    const start = (this.currentPage - 1) * subplotsPerPage;
+                    return this.allDistinctColumnValues.slice(start, start + subplotsPerPage);
+                } else {
+                    return null;
+                }
+            },
+            totalPages() {
+                return this.allDistinctColumnValues ? Math.ceil(this.allDistinctColumnValues.length / this.chartMetadata.subplots!.subplotsPerPage) : 1;
+            },
+            chartDataPage() {
+                if (this.pageDistinctColumnValues && this.chartData && this.chartMetadata.subplots) {
+                    const distinctColumn = this.chartMetadata.subplots.distinctColumn;
+                    const data = this.chartData["data"].filter((row: any) => this.pageDistinctColumnValues!.includes(row[distinctColumn]));
+                    return {...this.chartData, data}
+                } else {
+                    return this.chartData;
+                }
+            },
+            pageNumberText: function () {
+                return i18next.t("pageNumber", {
```

Comment:
Language works well but when changeLanguage is triggered selected tab "Time series" changes to Map

Reply:
Outside the scope of this ticket, should be covered by: https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-2612